.pre-commit-config.yaml

```
@@ -8,19 +8,16 @@ repos:
     -   id: end-of-file-fixer
     -   id: check-yaml
     -   id: check-added-large-files
-- repo: https://github.com/psf/black
-  rev: 24.8.0
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.8.3
   hooks:
-    - id: black
+    - id: ruff
+      args: [ --fix ]
+    - id: ruff-format
```

Much better, thank you! We can reduce noise by reverting these lines:
```suggestion
    - id: ruff-format
    - id: ruff
      args: [ --fix ]
```
While the end result is effectively the same, things that are fixed by `ruff-format` don't get reported as errors by `ruff` in this order, but they do in the other order.

---

Please read https://docs.astral.sh/ruff/integrations/#pre-commit
> When running with --fix, Ruff's lint hook should be placed ___before___ Ruff's formatter hook...

---

Fair enough. I guess it's a bit of a chicken and egg problem: ruff-format fixes things ruff complains about, but also stuff ruff --fix introduces, I guess I can live with the duplicate warnings.